ocf_data_sampler/load/satellite.py

```diff
@@ -1,59 +1,198 @@
+            raise TypeError(f"Coordinate {coord} should be {expected_kind}, not {actual_kind}")
+    
+    return data_array
+
+def _setup_optimal_environment():
+    """Apply optimization settings for cloud data streaming."""
+    logger.debug("Applying optimization settings for cloud streaming...")
+
+    os.environ["GCSFS_CACHE_TIMEOUT"] = "3600"
+    os.environ["GCSFS_BLOCK_SIZE"] = str(OPTIMAL_BLOCK_SIZE_MB * 1024 * 1024)
+    os.environ["GCSFS_DEFAULT_CACHE_TYPE"] = "readahead"
+    os.environ["GOOGLE_CLOUD_DISABLE_GRPC"] = "true"
 
-    ds = ds.rename(
+    dask.config.set(
         {
-            "variable": "channel",
-            "time": "time_utc",
-        },
+            "distributed.worker.memory.target": 0.7,
+            "array.chunk-size": "512MB",
+            "distributed.comm.compression": None,
+            "distributed.worker.threads": OPTIMAL_THREADS,
+        }
     )
+    logger.debug("Optimization environment configured successfully")
 
-    check_time_unique_increasing(ds.time_utc)
-    ds = make_spatial_coords_increasing(ds, x_coord="x_geostationary", y_coord="y_geostationary")
-    ds = ds.transpose("time_utc", "channel", "x_geostationary", "y_geostationary")
+def _parse_zarr_path(path: str) -> dict:
+    """Parse a path into its components, supporting both local and cloud paths."""
 
-    data_array = get_xr_data_array_from_xr_dataset(ds)
+    # Sol's recommended regex pattern - handles optional protocol and wildcards  
+    pattern = r"^(?:(?P<protocol>[\w]{2,6}):\/\/)?(?P<bucket>[\w\/-]+)\/(?P<prefix>[\w*.\/-]+?)(?:@(?P<sha1>[\w]+))?$"
+    match = re.match(pattern, path)
+    if not match:
+        raise ValueError(f"Invalid path format: {path}")
+    
+    components = match.groupdict()
+    
+    # Validation checks moved from match block
+    if components["sha1"] is not None and not components["prefix"].endswith(".icechunk"):
+        raise ValueError("Commit syntax (@commit) not supported for non-icechunk stores")
+    
+    if components["protocol"] == "gs" and components["prefix"] is not None and "*" in components["prefix"]:
+        raise ValueError("Wildcard (*) paths are not supported for GCP (gs://) URLs")
+    
+    return components
 
-    # Validate data types directly loading function
-    if not np.issubdtype(data_array.dtype, np.number):
-        raise TypeError(f"Satellite data should be numeric, not {data_array.dtype}")
+def _open_sat_data_icechunk(
+    protocol: str | None, bucket: str, prefix: str, sha1: str | None
+) -> xr.Dataset:
+    """Open satellite data from an Ice Chunk repository with optimized settings."""
+    
+    # Handle local icechunk paths
+    if protocol == "local":
+        # Remove @commit from path if present for repository opening
+        base_path = prefix.split('@')[0] if '@' in prefix else prefix
+        logger.info(f"Opening local Ice Chunk repository: {base_path}")
+        
+        storage = icechunk.local_filesystem_storage(base_path)
```

Comment:
Also, you shouldn't need this split, as the commit is already captured seperately to the path in the `sha1` variable. So it should be as straightforward as 

```python
storage = icechunk.local_filesystem_storage(prefix)
```